### PR TITLE
Login functionality

### PR DESCRIPTION
**Pull request by v74c63t**
login-func
Both login and google login work. I'm just waiting on the auth context provider to save the user id/username from login and for some changes to be made so google login can also provide that information to be saved, but I just wanted to get this up for now. I do have the userId and username saved in an object on a successful regular login, but it is not used for anything right now. Currently on a successful login, it just redirects to a placeholder dashboard page I created. This expects frontend to run on port 3000 and backend to run on 3001. Some adjustments were made to google auth so the GOOGLE_REDIRECT env variable need to be updated (localhost:3001 instead of localhost:3000).

### CODE REVIEW

File: frontend/src/Pages/Login Page/LoginPage.jsx [hunk 1]
```diff
@@ -16,24 +16,34 @@ function LoginPage () {
...
+        password: password
+      } 
+      // send username and password to login func in apiClient
+      // apiClient sends a req to backend login endpoint to check if username and password matches an existing user in the db
+      // if it matches, user is navigated to dashboard
+      apiClient.login(loginInfo).then( res => {
+        // console.log('data ', res)
+        if(res.data.success) { 
+          console.log('userId ', res.data.userId)
+          console.log('username ', res.data.username)
```
Comment (DorisSanchez-Velasquez) [inline frontend/src/Pages/Login Page/LoginPage.jsx:33]: Looks Great! Once auth context PR is merged, just make a quick hot fix to save the info into the user object :)

File: frontend/src/Pages/Login Page/LoginPage.jsx [hunk 2]
```diff
@@ -84,21 +80,22 @@ function LoginPage () {
...
-                        <path fill="#EA4335" d="M24 9.5c3.54 0 6.71 1.22 9.21 3.6l6.85-6.85C35.9 2.38 30.47 0 24 0 14.62 0 6.51 5.38 2.56 13.22l7.98 6.19C12.43 13.72 17.74 9.5 24 9.5z"></path>
-                        <path fill="#4285F4" d="M46.98 24.55c0-1.57-.15-3.09-.38-4.55H24v9.02h12.94c-.58 2.96-2.26 5.48-4.78 7.18l7.73 6c4.51-4.18 7.09-10.36 7.09-17.65z"></path>
-                        <path fill="#FBBC05" d="M10.53 28.59c-.48-1.45-.76-2.99-.76-4.59s.27-3.14.76-4.59l-7.98-6.19C.92 16.46 0 20.12 0 24c0 3.88.92 7.54 2.56 10.78l7.97-6.19z"></path>
-                        <path fill="#34A853" d="M24 48c6.48 0 11.93-2.13 15.89-5.81l-7.73-6c-2.15 1.45-4.92 2.3-8.16 2.3-6.26 0-11.57-4.22-13.47-9.91l-7.98 6.19C6.51 42.62 14.62 48 24 48z"></path>
-                        <path fill="none" d="M0 0h48v48H0z"></path>
+              <button className={LoginStyles['gsi-material-button']}>
+                {/* send user to google user flow for signing in if succesful user is navigated to dashboard */}
+                <a href={DEVELOPMENT_API_BASE_URL+'/login/auth/google'}>
```
Comment (DorisSanchez-Velasquez) [inline frontend/src/Pages/Login Page/LoginPage.jsx:85]: Looks awesome but just requesting a small change to API_BASE_URL instead; API_BASE_URL refers to a general url regardless of whether the site is in production or development mode while the other urls like DEVELOPMENT ... and PRODUCTION ... are more specific to that. That way, when changes are made, we don't have individually go in the files switching the urls when swtiching between dev and prod :)

**Review by DorisSanchez-Velasquez — CHANGES_REQUESTED** [review]
This looks great! Small changes requested for the url in the login page and quick hotfix to functionality once context PR is merged :)

File: frontend/src/Pages/Login Page/LoginPage.jsx (hunk 2 above)
Comment (v74c63t) [reply frontend/src/Pages/Login Page/LoginPage.jsx]: Fixed

**Review by DorisSanchez-Velasquez — APPROVED** [review]
Looks awesome!!

**Review by ItsJonasL — APPROVED** [review]
Looks great to me! Thanks for the hard work

**Pull request merged**